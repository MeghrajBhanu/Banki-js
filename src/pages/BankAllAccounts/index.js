import React, { useState, useEffect } from "react";

import { useDispatch, useSelector } from "react-redux";
import { useParams } from "react-router-dom";
import { create_account } from "../../redux/actions/createAccount";
import { clearMessage } from "../../redux/actions/message";
import { get_all_bankAccounts } from "../../redux/actions/pan_actions";
import image_map from "../../utils/images";
import classes from "./index.module.css";

/**
 * BankAllAccounts Page Shows all the Accounts user has in the particular bank.
 * @returns {React.FunctionComponent}
 */

const BankAllAccounts = () => {
  let { bankName } = useParams();
  const [errors, setError] = useState({ Balance: null, FixedDeposits: null });
  let user = useSelector((state) => state.auth.user);
  
  const { message } = useSelector((state) => state.message);
  const [valuess, setValues] = useState({
    AccountType: "",
    Balance: 500,
    FixedDeposits: 0,
  });
  useEffect(()=>{
    dispatch(clearMessage())
    
  },[])
  const handleInputValidation = (event) => {
    let message = "";
    if (event.target.name === "Balance") {
      if (event.target.value > 1000)
        message = "Balance should be less than 1000";
      else if (event.target.value === "") message = "Balance is required";
      else if (event.target.value < 500) message = "Minimum Balnce is 500";
      else message = null;
    }
    if (event.target.name === "FixedDeposits") {
      if (event.target.value > 5)
        message = "FixedDeposits should be less than 6";
      else if (event.target.value === "") message = "FixedDeposits is required";
      else if (event.target.value < 0)
        message = "FixedDeposits can't be Negative";
      else message = null;
    }
    
    setError((prev) => {
      return {
        ...prev,
        [event.target.name]: message,
      };
    });
  };
  const handleSubmit=()=>{
    dispatch(create_account({...valuess,pancard:user.pancard,name:user.name,bankName:bankName,email:user.email}))
    alert("Account created")
    }
  
  const handleChange = (e) => {
    if (e.target.name === "Salary" || e.target.name === "Savings") {
      setValues({ ...valuess, ["AccountType"]: e.target.value });
    } else {
      setValues({ ...valuess, [e.target.name]: e.target.value });
      handleInputValidation(e);
    }
  };
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(get_all_bankAccounts(user["pancard"], bankName));
  }, []);
  const data = useSelector((state) => state.pan.data);
  const isLoading = useSelector((state) => state.pan.isLoading);
  let src = image_map[bankName];
  return (
    <>
      {!isLoading && (
        <>
       
          <div className={classes["wrap"]}>
            <div className={classes["_8MlDE"]}>
              <img className={classes["H5I6J"]} src={src} alt="bankName" />
              <div className={classes["content"]}>
                <h2 className={classes["text-lead"]}>{bankName}</h2>
                <p className={classes["second-text"]}>India</p>
                <p className={classes["second-text"]}>Hyderabad,Telangana</p>
              </div>

              <h3 className={classes["set"]}>
                Total Bank Accounts: {data?.length}
              </h3>
            </div>
          </div>
          <div
            style={{
              backgroundColor: "seashell",
              width: "100%",
              height: "50px",
            }}
          >
            <button
              type="button"
              className="btn btn-secondary"
              data-bs-toggle="modal"
              data-bs-target="#exampleModal"
              style={{ float: "right", margin: "1px" }}
            >
              Add Account
            </button>
          </div>
          <table class="table mt-4">
            <thead class="thead-dark">
              <tr>
                <th scope="col">#Account No</th>
                <th scope="col">Account Type</th>
                <th scope="col">Fixed Deposits</th>
                <th scope="col">Balance</th>
              </tr>
            </thead>
            <tbody>
              {data?.map((item, i) => {
                return (
                  <tr>
                    <th scope="row">{item._id}</th>

                    <td>{item.AccountType}</td>
                    <td>{item.FixedDeposits}</td>
                    <td>{item.Balance}</td>
                  </tr>
                );
              })}
            </tbody>
          </table>
          <div
            className="modal fade"
            id="exampleModal"
            tabIndex="-1"
            role="dialog"
            aria-labelledby="exampleModalLabel"
            aria-hidden="true"
          >
            <div className="modal-dialog" role="document">
              <div className="modal-content">
                <div className="modal-header">
                  <h5 className="modal-title" id="exampleModalLabel">
                    Add Account
                  </h5>
                  <button
                    type="button"
                    className="close"
                    data-bs-dismiss="modal"
                    aria-label="Close"
                    onClick={() => {
                      setValues({
                        AccountType: "",
                        Balance: 500,
                        FixedDeposits: 0,
                      });
                      setError({ Balance: null, FixedDeposits: null });
                    }}
                  >
                    <span aria-hidden="true">&times;</span>
                  </button>
                </div>
                <div className="modal-body">
                  <form className="mx-1 mx-md-4">
                    <div className="d-flex flex-row align-items-center mb-4">
                      <i className="mr-2" style={{ marginRight: "10px" }}>
                        Account Type:{" "}
                      </i>

                      <div
                        aria-label="Account type"
                        name="Account type"
                        onChange={handleChange}
                      >
                        <input
                          type="radio"
                          value="Salary"
                          name="Salary"
                          checked={valuess.AccountType === "Salary"}
                        />{" "}
                        Salary
                        <span className="mx-3" />
                        <input
                          type="radio"
                          value="Savings"
                          name="Savings"
                          checked={valuess.AccountType === "Savings"}
                          defaultChecked
                        />{" "}
                        Savings
                      </div>
                    </div>

                    <div className="d-flex flex-row align-items-center mb-4">
                      <i className="mx-4">Balance: </i>

                      <div className="form-outline flex-fill mb-0">
                        <input
                          type="number"
                          aria-label="Balance"
                          name="Balance"
                          id="form3Example4c"
                          className="form-control"
                          placeholder="Balance"
                          value={valuess.Balance}
                          onChange={handleChange}
                          onBlur={handleInputValidation}
                          required="required"
                        />
                      </div>
                    </div>
                    {errors.Balance !== null ? (
                      <div
                        className="text-danger mb-3"
                        style={{ marginLeft: "120px" }}
                      >
                        {errors.Balance}
                      </div>
                    ) : null}
                    <div className="d-flex flex-row align-items-center mb-3">
                      <i className="mr-2" style={{ marginRight: "20px" }}>
                        Fixed Deposits:{" "}
                      </i>
                      <div className="form-outline flex-fill mb-0">
                        <input
                          type="number"
                          aria-label="FixedDeposits"
                          name="FixedDeposits"
                          value={valuess.FixedDeposits}
                          id="form3Example3c"
                          className="form-control"
                          placeholder="FixedDeposits"
                          onChange={handleChange}
                          onBlur={handleInputValidation}
                          required="required"
                        />
                      </div>
                    </div>
                    {errors.FixedDeposits !== null ? (
                      <div
                        className="text-danger mb-3"
                        style={{ marginLeft: "120px" }}
                      >
                        {errors.FixedDeposits}
                      </div>
                    ) : null}
                  </form>
                </div>
                
                <div className="modal-footer">
                  <button
                    type="button"
                    className="btn btn-secondary"
                    data-bs-dismiss="modal"
                    onClick={() => {
                      setValues({
                        AccountType: "",
                        Balance: 500,
                        FixedDeposits: 0,
                      });
                      setError({ Balance: null, FixedDeposits: null });
                    }}
                  >
                    Close
                  </button>
                  <button
                    type="button"
                    className="btn btn-primary"
                    data-bs-dismiss="modal"
                    onClick={handleSubmit}
                  >
                    Save changes
                  </button>
                </div>
              </div>
            </div>
          </div>
        </>
      )}
    </>
  );
};
export default BankAllAccounts;
