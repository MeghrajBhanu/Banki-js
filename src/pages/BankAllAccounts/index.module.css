.wrap{
    position: sticky;
    top: 0;
    z-index: 2;
    height: 245px;
    pointer-events: none;
    margin-bottom: 5px;
}
._8MlDE {
    background-color: #171a29;
    color: #fff;
    height: 245px;
    transition: transform .2s ease-in-out,background-color .2s ease-in-out,color .2s ease-in-out;
    -ms-transform-origin: left top;
    transform-origin: left top;
    -ms-transform: scaleX(1);
    transform: scaleX(1);
    pointer-events: auto;
    
    display:flex;
    justify-content: space-around;
}
.second-text{
    font-weight: lighter;
    opacity: 0.5;
}
.set{
    margin-top: auto;
    margin-bottom:auto;
}
.text-lead{
    color:white;
    font-weight: bolder;
}
._3TBoD {
    -ms-transform-origin: left top;
    transform-origin: left top;
    -ms-transform: scaleX(1);
    transform: scaleX(1);
    transition: transform .2s ease-in-out;
    display: table;
    height: 100%;
    table-layout: fixed;
    max-width: 300px;
    min-width: 300px;
    position: relative;
    margin: 0 auto;
}
._3sCWI {
    height: 165px;
    position: absolute;
    top: 0;
    opacity: 1;
    -ms-transform-origin: left top;
    transform-origin: left top;
    transform: translateZ(0) scaleX(1);
    transition: transform .2s ease-in-out;
}
._3mJdF {
    width: 254px;
    max-width: 254px;
    overflow: hidden;
    display: table-cell;
    vertical-align: top;
}
.H5I6J {
    width: 250px;
    height: 165px;
    display: inline-block;
    position: relative;
    overflow: hidden;
    background-color: #eef0f5;
    transform: translate3d(0,40px,0) scaleX(1);
    -ms-transform-origin: left top;
    transform-origin: left top;
    transition: transform .2s ease-in-out;
}
.content{
    margin-top: auto;
    margin-bottom: auto;
    color:white;
    display:flex;
    flex-direction:column;
}
._2Fixt {
    display: table-cell;
    vertical-align: middle;
    height: 100%;
    padding: 0 55px;
    transform: translateZ(0);
    transition: transform .2s ease-in-out;
    position: relative;
}
._1WDSQ {
    overflow: visible;
    position: relative;
    pointer-events: auto;
}
.U-jcg {
    
    display: flex;
    justify-content: space-between;
}
.OEfxz {
    display: flex;
}
._3aqeL {
    max-width: 538px;
    margin: 0;
    font-weight: 300;
    font-size: 32px;
}
._2cMZ_ {
    color: inherit;
    font-size: 14px;
    font-weight: 300;
    position: absolute;
    transition: opacity .2s ease-in-out;
    opacity: 0;
    top: 47px;
    width: 100%;
    z-index: -1;
}
._20F32 {
    display: inline-block;
}
._1BpLF {
    opacity: 1;
    transition: opacity .2s ease-in-out;
}
.Z4sK8 {
    color: inherit;
    width: 298px;
    padding: 74px 0 48px;
    height: 100%;
    
    display: table-cell;
    vertical-align: top;
}
._2FyFZ {
    transform: translateZ(0);
    transition: transform .2s ease-in-out;
    border: 1px solid #bec0c8;
    position: relative;
}
.QWCzK {
    color: inherit;
    font-size: 20px;
    font-weight: 500;
    background-color: #171a29;
    display: inline-block;
    padding-right: 10px;
    padding-bottom: 10px;
    text-transform: uppercase;
    position: absolute;
    top: -13px;
    left: -6px;
}
._3F2Nk {
    padding: 31px 40px 25px 25px;
}
.DM5zR {
    z-index: -1;
    line-height: 17px;
    display: -ms-flexbox;
    display: flex;
    -ms-flex-align: center;
    align-items: center;
    margin-top: 16px;
}

    
