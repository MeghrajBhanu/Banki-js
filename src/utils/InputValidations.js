export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}