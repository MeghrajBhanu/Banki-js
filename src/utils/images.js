import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;