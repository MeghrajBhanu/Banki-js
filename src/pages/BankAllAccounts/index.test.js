import { render, screen, waitFor } from "@testing-library/react";
import { onlywrap, renderComponent } from "../../utils/test_helper";
import renderer from "react-test-renderer";
import BankAllAccounts from ".";

describe("BankAllAccounts", () => {
  test("snapshot test", () => {
    const tree = renderer.create(onlywrap(<BankAllAccounts />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});