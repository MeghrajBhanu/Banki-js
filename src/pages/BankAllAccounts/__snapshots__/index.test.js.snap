// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankAllAccounts snapshot test 1`] = `null`;
